src/argilla/server/contexts/datasets.py

```diff
@@ -866,6 +879,8 @@ async def update_records(
                 include=RecordIncludeParam(keys=[RecordInclude.vectors], vectors=None),
             )
             await dataset.awaitable_attrs.vectors_settings
+            # This must be done to be sure that the engine index the updated vectors property
+            await refresh_records(records)
             await search_engine.index_records(dataset, records)
```

Comment:
As said before we can do:

```suggestion
        await search_engine.index_records(
            dataset, 
            await _preload_records_associations(records)
        )
```

If you don't want to do that in any case you can change it to:


```suggestion
            await _preload_records_associations(records)
            await search_engine.index_records(dataset, records)
```